presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/ResourceGroupRecord.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups.db;
+
+import com.facebook.presto.resourceGroups.ResourceGroupNameTemplate;
+import com.facebook.presto.resourceGroups.ResourceGroupSpec;
+import com.facebook.presto.spi.resourceGroups.SchedulingPolicy;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import org.skife.jdbi.v2.StatementContext;
+import org.skife.jdbi.v2.tweak.ResultSetMapper;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class ResourceGroupRecord
+{
+    private final String fullName;
+    private final ResourceGroupSpec resourceGroupSpec;
+    private final List<ResourceGroupRecord> subGroups;
+
+    private ResourceGroupRecord(
```

Comment:
Looks like this is mostly a wrapper for `ResourceGroupSpec` that also has the full name, and has `equals` and `hashCode`. How about implementing `equals` and `hashCode` on `ResourceGroupSpec` and then removing all the extra getters on this class and just having a `getResourceGroupSpec()`?